packages/odyssey-react-mui/src/theme/ThemeContrastProvider.test.tsx

```diff
@@ -0,0 +1,158 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { render, screen, act } from "@testing-library/react";
+import { renderHook } from "@testing-library/react-hooks";
+import {
+  ThemeContrastProvider,
+  useThemeContrastContext,
+  getBackgroundColor,
+} from "../ThemeContrastProvider";
+import * as Tokens from "@okta/odyssey-design-tokens";
+import "@testing-library/jest-dom";
+
+jest.mock("../ThemeContrastProvider", () => ({
+  ...jest.requireActual("../ThemeContrastProvider"),
+  getBackgroundColor: jest.fn(),
+}));
+
+const TestComponent = () => {
+  const { contrastMode, parentBackgroundColor } = useThemeContrastContext();
+  return (
+    <div role="status" aria-label="Contrast Mode">
+      {contrastMode}:{parentBackgroundColor}
+    </div>
+  );
+};
+
+describe("ThemeContrastProvider", () => {
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+    jest.resetAllMocks();
+  });
+
+  it("provides lowContrast mode by default when background is white", async () => {
+    (getBackgroundColor as jest.Mock).mockReturnValue("#ffffff");
+
+    render(
+      <ThemeContrastProvider>
+        <TestComponent />
+      </ThemeContrastProvider>,
+    );
+
+    await act(async () => {
+      jest.runAllTimers();
+    });
+
+    expect(
+      screen.getByRole("status", { name: "Contrast Mode" }),
+    ).toHaveTextContent("lowContrast:");
+  });
+
+  it("respects explicitly set contrast mode", async () => {
+    render(
+      <ThemeContrastProvider contrastMode="highContrast">
+        <TestComponent />
+      </ThemeContrastProvider>,
+    );
+
+    await act(async () => {
+      jest.runAllTimers();
+    });
+
+    expect(
+      screen.getByRole("status", { name: "Contrast Mode" }),
+    ).toHaveTextContent("highContrast:");
+  });
+
+  it("maintains lowContrast mode when background color is HueNeutral50", async () => {
+    (getBackgroundColor as jest.Mock).mockReturnValue(Tokens.HueNeutral50);
+
+    render(
+      <ThemeContrastProvider>
+        <TestComponent />
+      </ThemeContrastProvider>,
+    );
+
+    await act(async () => {
+      jest.runAllTimers();
+    });
+
+    expect(
+      screen.getByRole("status", { name: "Contrast Mode" }),
+    ).toHaveTextContent("lowContrast:");
+  });
+
+  it("maintains contrast mode when background color changes", async () => {
+    (getBackgroundColor as jest.Mock).mockReturnValue("#ffffff");
+
+    const { rerender } = render(
+      <ThemeContrastProvider>
+        <TestComponent />
+      </ThemeContrastProvider>,
+    );
+
+    await act(async () => {
+      jest.runAllTimers();
+    });
+
+    expect(
+      screen.getByRole("status", { name: "Contrast Mode" }),
+    ).toHaveTextContent("lowContrast:");
```

Comment:
I get why we have a `TestComponent` now, but I think there's a better way of doing these tests. Instead of having `TestComponent` with a role, we should just have it render `null`.

What we want to check isn't that the HTML output is `lowContrast`, we want to pass a callback `onContrast` to `TestComponent`, and we want to verify the value is correct.

## A better way
But... There's another way of doing that same thing which doesn't require adding random test components or callback functions.

We can use `ContrastModeContext.Consumer`, we can pass it a function as `children`, and capture that value in a variable. Then we can say `expect(contrastMode).toBe("lowContrast")`.

Now, it's type-safe, we know the expected values, and we're not adding extra code that could potentially interfere with our tests.

Reply:
Thinking about this more, we're essentially testing a hook (all that logic) and testing the 

Testing Library has a separate library for testing React hooks. It allows you to "render" the hook and ensure, given certain values, you get a specific outcome. This way, you can test the hook outside of the context provider component, and it should mean you don't have to do such extensive component testing since the hook holds all the logic.

Reply:
Point is, the way we're testing this today is super hacky and isn't going to catch everything because we're mocking the `getBackgroundColor` function leaving that completely untested.

And then we're adding more code that could cause us to not test correctly and receive a correct value when it's not correct.